css-inline/src/lib.rs

```diff
@@ -299,7 +310,9 @@ impl<'a> CSSInliner<'a> {
                 .attributes
                 .try_borrow_mut()
             {
-                if let Some(existing_style) = attributes.get_mut("style") {
+                if self.options.styles_as_attributes {
```

Comment:
Here we'd have a similar situation to the branchs below:
- If there is no "style" attribute, then go over `styles` (similar to line 319) and call `attributes.insert` with rule name / value.
- Otherwise, the existing "style" attribute should be popped from `attributes`, merged with collected styles, and then those "final" styles should be inserted into `attributes`

Reply:
Okay so within the conditional branch I created, we will have 2 conditional branches?

Reply:
Yep

Reply:
Okay so I think I got the first bullet point completed, I'm not quite sure if I understand the second one yet, but it's also a bit late for me haha. Will take a look tomorrow.

Reply:
Awesome! I think we can use `attribute.remove` straight away instead of `get_mut` (it returns Option<Attribute>), so the output doesn't have the "style" attribute.

For the second point, the workflow would look like the other branch with `attributes.insert`, but the existing "style" value should be merged together with ones collected from "style" tags (or other places). 

At the moment, `merge_styles` outputs `Result<String>` which is not exactly what is needed. Instead, there could be a new function that will insert new attributes into `attributes` instead of pushing them to a string.

Something like this

```rust
fn insert_as_attributes(
  existing_style: &str, 
  new_styles: &AHashMap<String, (Specificity, String)>, 
  attributes: &mut Attributes
) -> Result<()> {
    let mut input = cssparser::ParserInput::new(existing_style);
    let mut parser = cssparser::Parser::new(&mut input);
    let declarations = cssparser::DeclarationListParser::new(&mut parser, parser::CSSDeclarationListParser);
    let mut buffer: SmallVec<[String; 8]> = smallvec![];
    for declaration in declarations {
        let (name, value) = declaration?;
        attributes.insert(name, value);
        buffer.push(name.to_string());
    }
    for (property, (_, value)) in new_styles {
        if !buffer.contains(property) {
            attributes.insert(property, value)
        }
    }
    Ok(())
}
```

And the last branch would look like this:

```rust
insert_at_attributes(&existing_style.value, &styles, &mut attributes)
```

I didn't test these code samples, but they should be alright :) The main idea is that existing "style" attribute styles are inserted as separate attributes immediately as they have the maximum priority, then the other ones are inserted only if they aren't clashing with already inserted ones.

Then, there is a small corner case with quoting the value (see the `replace_double_quotes` macro). Before inserting, each property name & value should be checked:

```rust
if name.starts_with("font-family") && memchr::memchr(b'"', value.as_bytes()).is_some() {
    attributes.insert(name, &value.replace('"', "\'"))
} else {
    attributes.insert(name, value)
};
```

which could be also a small macro (the naming probably isn't much important - maybe `insert_unquoted`?)

Then, I think a few tests would be nice to have :) And a changelog entry